chihacknight/state-by-state-decarbonization
# power plant data import

## Overview

@joyceohiri, @hkier and I worked on reviewing the power plant data found in #23 and started on the import process last night. I wrapped it up today. This pull request adds the list of power plants for each state by category (coal, gas, oil).

Each power plant has the following attributes:

```json
{
    "plant_name": "Joliet 29",
    "fossil_fuel_category": "GAS",
    "county": "Will",
    "capacity_mw": "1,320.0",
    "utility_name": "Midwest Generations EME LLC"
},
```

Made some small changes to the `json_data_builder` utility function to allow for the internal `array_key` to be passed in (but defaulting to `emissionsByYear`. Also added some logic to handle pandas `NaN` values when outputting the json object.

Also added a line to convert state abbreviations to our full name in `get_and_clean_csv`

Finally, updated the `state-details.js` page to query this new data and create grouped lists for `coal_plants`, `gas_plants` and `oil_plants`

Closes #49 

## Testing Instructions

* run the data processing script and confirm data is output as expected ([instructions](https://github.com/chihacknight/state-by-state-decarbonization#data-processing))
* on the state detail page, confirm power plant data is available and being logged to the console:
![Screen Shot 2022-04-13 at 8 46 00 PM](https://user-images.githubusercontent.com/919583/163297654-f9efb094-4397-4909-84af-eb700b59f5d6.png)
